core/src/toga/widgets/mapview.py

```diff
@@ -0,0 +1,232 @@
+        self._title = title
+        self._subtitle = subtitle
+
+        # A pin isn't tied to a map at time of creation.
+        self.interface = None
+        self._native = None
+
+    def __repr__(self):
+        if self.subtitle:
+            label = f"; {self.title} - {self.subtitle}"
+        else:
+            label = f"; {self.title}"
+
+        return f"<MapPin @ {self.location}{label}>"
+
+    @property
+    def location(self) -> toga.LatLng:
+        "The (latitude, longitude) where the pin is located."
+        return self._location
+
+    @location.setter
+    def location(self, coord: toga.LatLng | tuple[float, float]) -> None:
+        self._location = toga.LatLng(*coord)
+        if self.interface:
+            self.interface._impl.update_pin(self)
+
+    @property
+    def title(self) -> str:
+        """The title of the pin."""
+        return self._title
+
+    @title.setter
+    def title(self, title: str) -> None:
+        self._title = str(title)
+        if self.interface:
+            self.interface._impl.update_pin(self)
+
+    @property
+    def subtitle(self) -> str | None:
+        "The subtitle of the pin."
+        return self._subtitle
+
+    @subtitle.setter
+    def subtitle(self, subtitle: str | None) -> None:
+        if subtitle is not None:
+            subtitle = str(subtitle)
+        self._subtitle = subtitle
+        if self.interface:
+            self.interface._impl.update_pin(self)
+
+
+class MapPinSet:
+    def __init__(self, interface, pins):
+        self.interface = interface
+        self._pins = set()
+
+        if pins:
+            for item in pins:
+                self.add(item)
+
+    def __repr__(self):
+        return f"<MapPinSet ({len(self)} pins)>"
+
+    def __iter__(self):
+        """Return an iterator over the pins on the map."""
+        return iter(self._pins)
+
+    def __len__(self):
+        """Return the number of pins being displayed."""
+        return len(self._pins)
+
+    def add(self, pin):
+        """Add a new pin to the map.
+
+        :param pin: The :any:`toga.MapPin` instance to add.
+        """
+        pin.interface = self.interface
+        self._pins.add(pin)
+        self.interface._impl.add_pin(pin)
+
+    def remove(self, pin):
+        """Remove a pin from the map.
+
+        :param pin: The  :any:`toga.MapPin` instance to remove.
+        """
+        self.interface._impl.remove_pin(pin)
+        self._pins.remove(pin)
+        pin.interface = None
+
+    def clear(self):
+        """Remove all pins from the map."""
+        for pin in self._pins:
+            self.interface._impl.remove_pin(pin)
+        self._pins = set()
+
+
+class OnSelectHandler(Protocol):
+    def __call__(self, widget: MapView, *, pin: MapPin, **kwargs: Any) -> None:
+        """A handler that will be invoked when the user selects a map pin.
+
+        .. note::
+            ``**kwargs`` ensures compatibility with additional arguments
+            introduced in future versions.
```

Comment:
I think it's clearer to document this using a `:param:` line, as in Canvas.OnTouchHandler.

Reply:
Good point; there's a couple of other handlers that use the ``..note::`` pattern that we could update.